compiler/runtime/obstack.c

```diff
@@ -197,19 +196,19 @@ void SKIP_destroy_Obstack(sk_saved_obstack_t* saved) {
     saved_head = saved->head;
     saved_end = saved->end;
   }
-  while (saved_head < page || saved_head > end) {
-    char* tofree = page;
+  while (saved_head < (char*)page || saved_head > end) {
```

Comment:
Not new in this diff, but my understanding is that the comparisons between the various `head`s and `page` would be more accurately done with `page->contents` instead, since `head` should never be able to point into the part of a page before the start of the contents. Also, I think the while condition is trying to test saved_head ∉ [page,end) and `saved_head == end` can't end up happening only because `head` can't point to the beginning of a page that happens to get allocated adjacent. So I think checking `saved_head >= end` would be clearer, or even rewritten as `!(page <= saved_head && saved_head <= end)`.